src/main/java/mekanism/client/render/item/ChemicalFluidBarDecorator.java

```diff
@@ -0,0 +1,116 @@
+import com.mojang.blaze3d.vertex.BufferBuilder;
+import com.mojang.blaze3d.vertex.Tesselator;
+import mekanism.api.chemical.Chemical;
+import mekanism.api.chemical.ChemicalStack;
+import mekanism.api.chemical.IChemicalHandler;
+import mekanism.api.math.MathUtils;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.Font;
+import net.minecraft.client.renderer.entity.ItemRenderer;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.client.IItemDecorator;
+import net.minecraftforge.client.extensions.common.IClientFluidTypeExtensions;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidUtil;
+import net.minecraftforge.fluids.capability.IFluidHandlerItem;
+
+import java.util.Optional;
+import java.util.function.Predicate;
+
+public class ChemicalFluidBarDecorator implements IItemDecorator {
+
+    private final Capability<? extends IChemicalHandler<?,?>>[] chemicalCaps;
+    private final boolean showFluid;
+    private final Predicate<ItemStack> visibleFor;
+
+    /**
+     * @param showFluid if the fluidcapability should be checked for display, display above chemicalCaps if both are present
+     * @param visibleFor checks if bars should be rendered for the given itemstack
+     * @param chemicalCaps the capabilities to be displayed in order, starting from the bottom
+     */
+    public ChemicalFluidBarDecorator(boolean showFluid, Predicate<ItemStack> visibleFor, Capability<? extends IChemicalHandler<?,?>>... chemicalCaps) {
+        this.showFluid = showFluid;
+        this.chemicalCaps = chemicalCaps;
+        this.visibleFor = visibleFor;
+    }
+
+    @Override
+    public boolean render(Font font, ItemStack stack, int xOffset, int yOffset, float blitOffset) {
+        if (!visibleFor.test(stack))
+            return false;
+        RenderSystem.disableDepthTest();
+        RenderSystem.disableTexture();
+        RenderSystem.disableBlend();
+        Tesselator tesselator = Tesselator.getInstance();
+        BufferBuilder bufferBuilder = tesselator.getBuilder();
+        yOffset = yOffset + 12;
+        for (Capability<? extends IChemicalHandler<?,?>> chemicalCap : chemicalCaps) {
+            Optional<? extends IChemicalHandler<?, ?>> capabilityInstance = stack.getCapability(chemicalCap).resolve();
+            if (capabilityInstance.isPresent()) {
+                IChemicalHandler<?, ?> chemicalHandler = capabilityInstance.get();
+                if (chemicalHandler.getTanks() == 0) {
+                    continue;
+                }
+                int tank = 0;
+                if (chemicalHandler.getTanks() > 1 && Minecraft.getInstance().level != null) {
+                    //Cycle through multiple tanks every second, to save some space if multiple tanks are present
+                    tank = (int) (Minecraft.getInstance().level.getGameTime() / 20) % chemicalHandler.getTanks();
```

Comment:
This feels kind of weird, though I can understand at some point we would run out of space... though maybe it is worth it. Either way I think it would be worth just having a helper method like:
```java
private static int getDisplayTank(int tanks) {
    if (tanks == 0) {
        return -1;
    } else if (tanks > 1 && level != null) {
        //Cycle
        return (int) (level.getGameTime() / 20) % tanks;
    }
    return 0;
}
```
And then effectively change the code for this and fluids to be:
```java
int tank = getTank(...getTanks());
if (tank != 1) {
    renderBar....
    yOffset--;
}
```